### DISCUSSION TITLE
ENUM values change in schemas not reported in .md file.

### DISCUSSION THREAD

**Discussion by dattadebarshi** [OpenAPITools/openapi-diff#520] [General]
I have compared 2 API Specifications json. The Specification contains multiple component schemas. But what I observed is that if there is an Enum Array (same array) is present in different component schemas, then while publishing the diff for Rest Call only once the ENUM change is reported although multiple schema changes are applicable for that or other Rest Call with the same Enum Array list.
![ENUM Array list](https://github.com/OpenAPITools/openapi-diff/assets/108448771/2cd9274d-b0b8-4fd7-a5a0-f847af67b40b)
![image](https://github.com/OpenAPITools/openapi-diff/assets/108448771/6daaff21-df75-4031-8449-717d59f119cb)
![image](https://github.com/OpenAPITools/openapi-diff/assets/108448771/1fbc1424-45b1-4435-9b3b-63a1d3693dcf)

**Comment by joschi** [comment]
@dattadebarshi Could you please provide a minimized version of the OpenAPI specs to reproduce the issue?

**Reply by dattadebarshi** [reply]
New Spec
`{
  "openapi": "3.0.1",
  "paths": {
    "/check/api/Id/Exp{ExpId}": {
      "get": {
        "responses": {
          "default": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExpRO"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ExpRO": {
        "type": "object",
        "properties": {
          "data": {
            "$ref": "#/components/schemas/ExpDataRO"
          }
        }
      },
      "ExpDataRO": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string"
          },
          "type": {
            "type": "string",
            "enum": [
              "tpeExpectations"
            ]
          },
          "attributes": {
            "$ref": "#/components/schemas/ExpAttributesRO"
          }
        }
      },
      "ExpAttributesRO": {
        "type": "object",
        "properties": {
          "Terminations": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/TerminationRO"
            }
          }
        }
      },
      "TerminationRO": {
        "type": "object",
        "properties": {
          "NetworkConnectionPackage": {
            "$ref": "#/components/schemas/NetworkConnectionRO"
          },
          "ponConnectionPackage": {
            "$ref": "#/components/schemas/PonConnectionRO"
          },
          "datalinkPackage": {
            "$ref": "#/components/schemas/DatalinkRO"
          }
        }
      },
      "NetworkConnectionRO": {
        "type": "object",
        "properties": {
          "tagAction": {
            "type": "string",
            "enum": [
              "none",
              "pop",
              "push",
              "stamp"
            ]
          },
          "floodingLink": {
            "type": "string",
            "enum": [
              "enabled",
              "disabled"
            ]
          }
        }
      },
      "PonConnectionRO": {
        "type": "object",
        "properties": {
          "tagAction": {
            "type": "string",
            "enum": [
              "none",
              "pop",
              "push",
              "stamp"
            ]
          }
        }
      },
      "DatalinkRO": {
        "type": "object",
        "properties": {
          "adminState": {
            "type": "string",
            "enum": [
              "enabled",
              "disabled"
            ]
          }
        }
      }
    }
  }
}`

Old Spec
`{
  "openapi": "3.0.1",
  "paths": {
    "/check/api/Id/Exp{ExpId}": {
      "get": {
        "responses": {
          "default": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExpRO"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ExpRO": {
        "type": "object",
        "properties": {
          "data": {
            "$ref": "#/components/schemas/ExpDataRO"
          }
        }
      },
      "ExpDataRO": {
        "type": "object",
        "properties": {
          "id": {
            "type": "string"
          },
          "type": {
            "type": "string",
            "enum": [
              "tpeExpectations"
            ]
          },
          "attributes": {
            "$ref": "#/components/schemas/ExpAttributesRO"
          }
        }
      },
      "ExpAttributesRO": {
        "type": "object",
        "properties": {
          "Terminations": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/TerminationRO"
            }
          }
        }
      },
      "TerminationRO": {
        "type": "object",
        "properties": {
          "nniNetwork": {
            "$ref": "#/components/schemas/NetworkRO"
          },
          "ponConnection": {
            "$ref": "#/components/schemas/PonConnectionRO"
          }
        }
      },
      "NetworkRO": {
        "type": "object",
        "properties": {
          "tagAction": {
            "type": "string",
            "enum": [
              "none",
              "pop",
              "push",
              "stamp"
            ]
          },
          "floodingLink": {
            "type": "string",
            "enum": [
              "enabled",
              "disabled"
            ]
          }
        }
      },
      "PonConnectionRO": {
        "type": "object",
        "properties": {
          "tagAction": {
            "type": "string",
            "enum": [
              "none",
              "pop",
              "push",
              "stamp"
            ]
          }
        }
      }
    }
  }
}`

Diff md 

#### What's Changed
---

##### `GET` /check/api/Id/Exp{ExpId}


###### Return Type:

Changed response : **default **

* Changed content type : `application/json`

    * Changed property `data` (object)

        * Changed property `attributes` (object)

            * Changed property `Terminations` (array)

                Changed items (object):

                * Added property `NetworkConnectionPackage` (object)

                    * Property `tagAction` (string)

                        Enum values:

                        * `none`
                        * `pop`
                        * `push`
                        * `stamp`
                    * Property `floodingLink` (string)

                        Enum values:

                        * `enabled`
                        * `disabled`
                * Added property `ponConnectionPackage` (object)

                    * Property `tagAction` (string)

                * Added property `datalinkPackage` (object)

                    * Property `adminState` (string)

                * Deleted property `nniNetwork` (object)

                * Deleted property `ponConnection` (object)

@joschi Please find the Specs above. Here the enums are not reported datalinkPackage -> adminState and ponConnectionPackage -> tagAction

**Reply by dattadebarshi** [reply]
@joschi Any update on the above issue?

**Reply by dattadebarshi** [reply]
There is a similar issue with properties as well:
Old Spec Json

`{
  "openapi": "3.0.1",
  "paths": {
    "/check/API": {
      "get": {
        "responses": {
          "default": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExpAPIRO"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ExpAPIRO": {
        "type": "object",
        "properties": {
          "data": {
            "$ref": "#/components/schemas/ExpAPIDataRO"
          }
        }
      },
      "ExpAPIDataRO": {
        "type": "object",
        "properties": {
          "relationships": {
            "$ref": "#/components/schemas/ExpAPIRelationshipsRO"
          }
        }
      },
      "ExpAPIRelationshipsRO": {
        "type": "object",
        "properties": {}
      }
    }
  }
}`

New Spec JSON

`{
  "openapi": "3.0.1",
  "paths": {
    "/check/API": {
      "get": {
        "responses": {
          "default": {
            "content": {
              "application/json": {
                "schema": {
                  "$ref": "#/components/schemas/ExpAPIRO"
                }
              }
            }
          }
        }
      }
    }
  },
  "components": {
    "schemas": {
      "ExpAPIRO": {
        "type": "object",
        "properties": {
          "data": {
            "$ref": "#/components/schemas/ExpAPIDataRO"
          }
        }
      },
      "ExpAPIDataRO": {
        "type": "object",
        "properties": {
          "relationships": {
            "$ref": "#/components/schemas/ExpAPIRelationshipsRO"
          }
        }
      },
      "ExpAPIRelationshipsRO": {
        "type": "object",
        "properties": {
          "linkedTpes": {
            "$ref": "#/components/schemas/OneToManyRelationshipRO"
          },
          "shelfNetworkConstruct": {
            "$ref": "#/components/schemas/OneToOneRelationshipRO"
          }
        }
      },
      "OneToManyRelationshipRO": {
        "type": "object",
        "properties": {
          "data": {
            "type": "array",
            "items": {
              "$ref": "#/components/schemas/RelationshipDataRO"
            }
          },
          "meta": {
            "$ref": "#/components/schemas/RelationshipMetaDataRO"
          }
        }
      },
      "RelationshipDataRO": {
        "type": "object",
        "properties": {
          "type": {
            "type": "string"
          },
          "id": {
            "type": "string"
          },
          "attributes": {
            "$ref": "#/components/schemas/RelationshipAttributesRO"
          }
        }
      },
      "RelationshipAttributesRO": {
        "type": "object",
        "properties": {
          "additionalAttributes": {
            "$ref": "#/components/schemas/AdditionalAttributesRO"
          }
        }
      },
      "AdditionalAttributesRO": {
        "type": "object"
      },
      "RelationshipMetaDataRO": {
        "type": "object",
        "properties": {
          "partiallyPopulated": {
            "type": "boolean"
          }
        }
      },
      "OneToOneRelationshipRO": {
        "type": "object",
        "properties": {
          "data": {
            "$ref": "#/components/schemas/RelationshipDataRO"
          },
          "meta": {
            "$ref": "#/components/schemas/RelationshipMetaDataRO"
          }
        }
      }
    }
  }
}`


Diff MD: 

`#### What's Changed
---

##### `GET` /check/API


###### Return Type:

Changed response : **default **

* Changed content type : `application/json`

    * Changed property `data` (object)

        * Changed property `relationships` (object)

            * Added property `linkedTpes` (object)

                * Property `data` (array)

                    Items (object):

                    * Property `type` (string)

                    * Property `id` (string)

                    * Property `attributes` (object)

                        * Property `additionalAttributes` (object)

                * Property `meta` (object)

                    * Property `partiallyPopulated` (boolean)

            * Added property `shelfNetworkConstruct` (object)

                * Property `data` (object)

                * Property `meta` (object)

`
